dot/core/messages.go

```diff
@@ -32,42 +32,58 @@ func (s *Service) HandleTransactionMessage(msg *network.TransactionMessage) (boo
 	txs := msg.Extrinsics
 	var toPropagate []types.Extrinsic
 
-	rt, err := s.blockState.GetRuntime(nil)
+	head, err := s.blockState.BestBlockHeader()
+	if err != nil {
+		return false, err
+	}
+
+	hash := head.Hash()
+	rt, err := s.blockState.GetRuntime(&hash)
 	if err != nil {
 		return false, err
 	}
 
 	for _, tx := range txs {
-		ts, err := s.storageState.TrieState(nil)
-		if err != nil {
-			return false, err
-		}
+		err = func() error {
+			s.storageState.Lock()
```

Comment:
Should we move this lock outside of the loop?

Reply:
I figured to put it inside the loop since if it locks outside the loop instead of for each transaction, it might cause more lock contention